core/parser.mly

```diff
@@ -443,6 +445,12 @@ tlfunbinding:
 | OP OPERATOR pattern perhaps_location block                   { ((dl_unl, false), $2, [[$3]], $4, $5)          }
 | OP pattern OPERATOR perhaps_location block                   { ((dl_unl, false), $3, [[$2]], $4, $5)          }
 
+switch_tlfunbinding:
+| fun_kind VARIABLE arg_lists perhaps_location switch_body     { ($1, $2, $3, $4, $5)   }
+
+switch_body:
```

Comment:
Could you rename this to `switch_funlit_body`? Otherwise, it may be confused with the body of an actual switch statement by someone who is unfamiliar with the codebase.